httpx/middleware/reqsigssm/request_signing_ssm.go

```diff
@@ -0,0 +1,27 @@
+package reqsigssm
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/remind101/pkg/httpx/middleware"
+)
+
+// NewWithPath takes a path, and returns an instance of
+// middleware.StaticSigningKeyRepository initialized with client keys / secrets
+// fetched from the given path. The assumption is that path points to an SSM
+// directory containing the client keys and secrets.
+func NewWithPath(path string) (*middleware.StaticSigningKeyRepository, error) {
```

Comment:
May be nice to provide an initializer that takes an `client.ConfigProvider`. Sometimes, it's handy to configure an session instance globally, and inject it around:

```golang
New(config client.ConfigProvier, path string) (*middleware.StaticSigningKeyRepository, error)
NewWithPath(path string) (*middleware.StaticSigningKeyRepository, error)
```